spec/integrations/fixtures/rspec/before_all_custom_adapter_fixture.rb

```diff
@@ -5,13 +5,31 @@
 require "test_prof/recipes/rspec/before_all"
 
 module MyAdapter
+  @nested_transaction_count = 0
```

Comment:
We didn't use nested transactions if we didn't have nested example groups.
Now one transaction is open for an example group, and another for the example itself, so the test adapter needs to be capable of providing SAVEPOINT support to avoid the error when nested `BEGIN` is called.